sessionctx/stmtctx/stmtctx_test.go

```diff
@@ -96,49 +98,44 @@ func TestWeakConsistencyRead(t *testing.T) {
 	store, clean := testkit.CreateMockStore(t)
 	defer clean()
 
-	lastWeakConsistency := func(tk *testkit.TestKit) bool {
-		return tk.Session().GetSessionVars().StmtCtx.WeakConsistency
-	}
-
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists t")
 	tk.MustExec("create table t(id int primary key, c int, c1 int, unique index i(c))")
+
+	execAndCheck := func(sql string, rows [][]interface{}, isolationLevel kv.IsoLevel) {
```

Comment:
Will this check be executed for each kv request?

Reply:
The hook function will be executed in distsql only.